src/bot/commands/random/color-generator.ts

```diff
@@ -0,0 +1,106 @@
+import { Command, Input } from '@api';
+import { Emoji } from '@bot/libraries/emoji';
+import { RichEmbed } from 'discord.js';
+
+export class Color extends Command {
+    constructor() {
+        super({
+            name: 'color',
+            description: 'Generates a random color, or provides information about the provided color',
+            arguments: [
+                {
+                    name: 'color',
+                    expand: true
+                }
+            ]
+        });
+    }
+
+    async execute(input: Input) {
+        let color = input.getArgument('color') as string;
+
+        try {
+            let { r, g, b } = this.parse(color);
+            let hex = this.hex(r, g, b);
+            let url = `https://temp.bailey.sh/${r}/${g}/${b}/image.jpg`;
+
+            await input.channel.send(new RichEmbed({
+                author: {
+                    name: hex,
+                    icon_url: url
+                },
+                color: parseInt(hex.substring(1), 16)
+            }));
+        }
+        catch (err) {
+            await input.channel.send(`${Emoji.ERROR}  ${err.message}`);
+        }
+    }
+
+    private parse(input: string) : {r:number, g:number, b:number} {
+        let r : number = 0, g : number = 0, b : number = 0;
+        let rgbExp = /(\d{0,3})[\s;,]+(\d{0,3})[\s;,]+(\d{0,3})/;
+
+        if (!input) {
+            return {
+                r: _.random(0, 255),
+                g: _.random(0, 255),
+                b: _.random(0, 255)
+            };
+        }
+
```

Comment:
I don't think it's as clear as it should be that this is where the colors are randomized. Maybe rename this method to "parseOrRandomize", move everything inside the return statement into a "randomize" method, and move everything past it into a "parse" method? I could figure out that the randomization happens here anyways, so it's probably not a dire issue.